sr_speech_control/src/sr_speech_control/speech_control.py

```diff
@@ -54,22 +76,37 @@ def _recognizer_callback(self, recognizer, audio):
             return
 
         result = [str(x).lower() for x in result.split(' ')]
-        if len(result) > 1:
-            if self._filter_word(result[0], [self.trigger_word]) == self.trigger_word:
-                self.command_publisher.publish(' '.join([self._filter_word(x, self.command_words) for x in result[1:]]))
+
+        if self._filter_word(result[0], self.trigger_word) == self.trigger_word:
+            command = self._filter_word(''.join(result[1:]), self.command_words)
+            if command in self.command_words:
+                self.command_to_be_executed = command
 
     def _filter_word(self, word, dictionary, offset=0.5):
+        if word in self.similar_words_dict:
+            word = self.similar_words_dict[word]
+
         result = get_close_matches(word, dictionary, 1, offset)
         if not result:
             return word
         return result[0]
 
+    def run(self):
+        rospy.loginfo("Started speech control. Trigger word: {}".format(self.trigger_word))
+        while not rospy.is_shutdown():
+            if self.command_to_be_executed:
+                rospy.loginfo("Executing: {}.".format(self.command_to_be_executed))
+                self.command_publisher.publish(self.command_to_be_executed)
+                self.command_to_be_executed = None
+        self._stop_listening(wait_for_stop=False)
+
 
 if __name__ == "__main__":
-    rospy.init_node('sr_speech_control', anonymous=True)
+    rospy.init_node('example_speech_control', anonymous=True)
+
+    trigger_word = "shadow"
+    command_words = ["grasp", "release", "disable", "enable", "engage"]
```

Comment:
maybe it would be good to also have this in a yaml file, so it will be easier to modify?

Reply:
I'd rather not, that way anytime someone creates a new node they can just put that stuff there instead of having to create a new additional file. This is very simple format so I'd rather leave it within the node code.